tests/test_background_worker.py

```
@@ -0,0 +1,473 @@
+        mock_session.request.return_value = mock_response
+        mock_session_cls.return_value.__enter__.return_value = mock_session
+
+        job = {
+            'url': 'http://api.example.com/endpoint',
+            'headers': {'Authorization': 'Bearer token123', 'X-Custom': 'value'}
+        }
+
+        with patch('agents.background.worker._validate_and_resolve_target',
+                   return_value=(True, '', '1.2.3.4')):
+            with patch('agents.background.worker._domain_allowed', return_value=True):
+                result = process_http_job(job)
```

Ruff (0.14.11)

Local variable `result` is assigned to but never used

Remove assignment to unused variable `result`

(F841)